### ISSUE TITLE
error while trying to build on windows

### ISSUE THREAD

**Issue by TrevLite** [cinemast/libjson-rpc-cpp#105]
64bit machine running windows 64-bit with 32-bit cygwin
trying to build form source but get a dll error

I hope that this is the correct place to report errors.

$ make
Scanning dependencies of target jsonrpccommon
[  1%] Building CXX object src/jsonrpccpp/CMakeFiles/jsonrpccommon.dir/common/errors.cpp.o
[  3%] Building CXX object src/jsonrpccpp/CMakeFiles/jsonrpccommon.dir/common/exception.cpp.o
[  4%] Building CXX object src/jsonrpccpp/CMakeFiles/jsonrpccommon.dir/common/procedure.cpp.o
[  6%] Building CXX object src/jsonrpccpp/CMakeFiles/jsonrpccommon.dir/common/specificationparser.cpp.o
[  7%] Building CXX object src/jsonrpccpp/CMakeFiles/jsonrpccommon.dir/common/specificationwriter.cpp.o
[  9%] Linking CXX shared library ../../bin/cygjsonrpccpp-common-0.dll
[  9%] Built target jsonrpccommon
Scanning dependencies of target jsonrpcclient
[ 10%] Building CXX object src/jsonrpccpp/CMakeFiles/jsonrpcclient.dir/client/batchcall.cpp.o
[ 12%] Building CXX object src/jsonrpccpp/CMakeFiles/jsonrpcclient.dir/client/batchresponse.cpp.o
[ 13%] Building CXX object src/jsonrpccpp/CMakeFiles/jsonrpcclient.dir/client/client.cpp.o
[ 15%] Building CXX object src/jsonrpccpp/CMakeFiles/jsonrpcclient.dir/client/rpcprotocolclient.cpp.o
[ 16%] Building CXX object src/jsonrpccpp/CMakeFiles/jsonrpcclient.dir/client/connectors/httpclient.cpp.o
[ 18%] Linking CXX shared library ../../bin/cygjsonrpccpp-client-0.dll
/usr/local/lib/libjsoncpp.a(json_writer.cpp.o):json_writer.cpp:(.text+0x4d0): multiple definition of `Json::Writer::~Writer()'
../../lib/libjsonrpccpp-common.dll.a(d001165.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_writer.cpp.o):json_writer.cpp:(.text+0x4f0): multiple definition of `Json::FastWriter::FastWriter()'
../../lib/libjsonrpccpp-common.dll.a(d000915.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_writer.cpp.o):json_writer.cpp:(.text+0x540): multiple definition of `Json::StyledWriter::StyledWriter()'
../../lib/libjsonrpccpp-common.dll.a(d000959.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_writer.cpp.o):json_writer.cpp:(.text+0x2eb0): multiple definition of `Json::FastWriter::write(Json::Value const&)'
../../lib/libjsonrpccpp-common.dll.a(d000914.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_writer.cpp.o):json_writer.cpp:(.text+0x62a0): multiple definition of `Json::StyledWriter::write(Json::Value const&)'
../../lib/libjsonrpccpp-common.dll.a(d000955.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0xf50): multiple definition of `Json::Value::Value(Json::ValueType)'
../../lib/libjsonrpccpp-common.dll.a(d001094.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0xff0): multiple definition of `Json::Value::Value(int)'
../../lib/libjsonrpccpp-common.dll.a(d001102.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x10e0): multiple definition of `Json::Value::Value(char const*)'
../../lib/libjsonrpccpp-common.dll.a(d001095.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x1370): multiple definition of `Json::Value::Value(std::string const&)'
../../lib/libjsonrpccpp-common.dll.a(d001099.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x1520): multiple definition of `Json::Value::operator=(Json::Value)'
../../lib/libjsonrpccpp-common.dll.a(d001120.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x2fe0): multiple definition of `Json::Value::operator!=(Json::Value const&) const'
../../lib/libjsonrpccpp-common.dll.a(d001418.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x3210): multiple definition of `Json::Value::size() const'
../../lib/libjsonrpccpp-common.dll.a(d001374.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x3900): multiple definition of `Json::Value::operator[](std::string const&) const'
../../lib/libjsonrpccpp-common.dll.a(d001413.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x3b70): multiple definition of `Json::Value::isMember(char const*) const'
../../lib/libjsonrpccpp-common.dll.a(d001400.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x3ba0): multiple definition of `Json::Value::isMember(std::string const&) const'
../../lib/libjsonrpccpp-common.dll.a(d001402.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x3bd0): multiple definition of `Json::Value::isNull() const'
../../lib/libjsonrpccpp-common.dll.a(d001383.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x3bf0): multiple definition of `Json::Value::isInt() const'
../../lib/libjsonrpccpp-common.dll.a(d001379.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x40b0): multiple definition of `Json::Value::isString() const'
../../lib/libjsonrpccpp-common.dll.a(d001404.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x40c0): multiple definition of `Json::Value::isArray() const'
../../lib/libjsonrpccpp-common.dll.a(d001388.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x40d0): multiple definition of `Json::Value::isObject() const'
../../lib/libjsonrpccpp-common.dll.a(d001403.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x4700): multiple definition of `Json::Value::toStyledString() const'
../../lib/libjsonrpccpp-common.dll.a(d001366.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x6320): multiple definition of `Json::Value::asInt() const'
../../lib/libjsonrpccpp-common.dll.a(d001376.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x6ef0): multiple definition of `Json::Value::asString() const'
../../lib/libjsonrpccpp-common.dll.a(d001397.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x7780): multiple definition of `Json::Value::~Value()'
../../lib/libjsonrpccpp-common.dll.a(d001118.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x7ad0): multiple definition of `Json::Value::Value(Json::Value const&)'
../../lib/libjsonrpccpp-common.dll.a(d001098.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x9ed0): multiple definition of `Json::Value::operator[](unsigned int)'
../../lib/libjsonrpccpp-common.dll.a(d001125.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0xa950): multiple definition of `Json::Value::append(Json::Value const&)'
../../lib/libjsonrpccpp-common.dll.a(d001074.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0xb970): multiple definition of `Json::Value::operator[](char const*)'
../../lib/libjsonrpccpp-common.dll.a(d001121.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0xb9a0): multiple definition of `Json::Value::operator[](std::string const&)'
../../lib/libjsonrpccpp-common.dll.a(d001123.o):(.text+0x0): first defined here
collect2: error: ld returned 1 exit status
src/jsonrpccpp/CMakeFiles/jsonrpcclient.dir/build.make:201: recipe for target 'bin/cygjsonrpccpp-client-0.dll' failed
make[2]: *** [bin/cygjsonrpccpp-client-0.dll] Error 1
CMakeFiles/Makefile2:153: recipe for target 'src/jsonrpccpp/CMakeFiles/jsonrpcclient.dir/all' failed
make[1]: *** [src/jsonrpccpp/CMakeFiles/jsonrpcclient.dir/all] Error 2
Makefile:160: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by cinemast** [comment]
This is the right place. I will check on that.

**Comment by TrevLite** [comment]
Thank you.
I am using libjson-rpc-cpp for a school project but could not get it to compile on any of my computers.
I hope that you can fix this error.

**Comment by TrevLite** [comment]
here is the error message on my other computer

$ make
[  1%] Building CXX object src/jsonrpccpp/CMakeFiles/jsonrpccommon.dir/common/errors.cpp.o
[  3%] Building CXX object src/jsonrpccpp/CMakeFiles/jsonrpccommon.dir/common/exception.cpp.o
[  5%] Building CXX object src/jsonrpccpp/CMakeFiles/jsonrpccommon.dir/common/procedure.cpp.o
[  7%] Building CXX object src/jsonrpccpp/CMakeFiles/jsonrpccommon.dir/common/specificationparser.cpp.o
[  9%] Building CXX object src/jsonrpccpp/CMakeFiles/jsonrpccommon.dir/common/specificationwriter.cpp.o
Linking CXX shared library ../../bin/cygjsonrpccpp-common-0.dll
[  9%] Built target jsonrpccommon
[ 11%] Building CXX object src/jsonrpccpp/CMakeFiles/jsonrpcclient.dir/client/batchcall.cpp.o
[ 12%] Building CXX object src/jsonrpccpp/CMakeFiles/jsonrpcclient.dir/client/batchresponse.cpp.o
[ 14%] Building CXX object src/jsonrpccpp/CMakeFiles/jsonrpcclient.dir/client/client.cpp.o
[ 16%] Building CXX object src/jsonrpccpp/CMakeFiles/jsonrpcclient.dir/client/rpcprotocolclient.cpp.o
[ 18%] Building CXX object src/jsonrpccpp/CMakeFiles/jsonrpcclient.dir/client/connectors/httpclient.cpp.o
Linking CXX shared library ../../bin/cygjsonrpccpp-client-0.dll
/usr/local/lib/libjsoncpp.a(json_writer.cpp.o):json_writer.cpp:(.text+0x4d0): multiple definition of `Json::Writer::~Writer()'
../../lib/libjsonrpccpp-common.dll.a(d001165.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_writer.cpp.o):json_writer.cpp:(.text+0x4f0): multiple definition of `Json::FastWriter::FastWriter()'
../../lib/libjsonrpccpp-common.dll.a(d000915.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_writer.cpp.o):json_writer.cpp:(.text+0x540): multiple definition of `Json::StyledWriter::StyledWriter()'
../../lib/libjsonrpccpp-common.dll.a(d000959.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_writer.cpp.o):json_writer.cpp:(.text+0x2eb0): multiple definition of `Json::FastWriter::write(Json::Value const&)'
../../lib/libjsonrpccpp-common.dll.a(d000914.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_writer.cpp.o):json_writer.cpp:(.text+0x62a0): multiple definition of `Json::StyledWriter::write(Json::Value const&)'
../../lib/libjsonrpccpp-common.dll.a(d000955.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0xf50): multiple definition of `Json::Value::Value(Json::ValueType)'
../../lib/libjsonrpccpp-common.dll.a(d001094.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0xff0): multiple definition of `Json::Value::Value(int)'
../../lib/libjsonrpccpp-common.dll.a(d001102.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x10e0): multiple definition of `Json::Value::Value(char const*)'
../../lib/libjsonrpccpp-common.dll.a(d001095.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x1370): multiple definition of `Json::Value::Value(std::string const&)'
../../lib/libjsonrpccpp-common.dll.a(d001099.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x1520): multiple definition of `Json::Value::operator=(Json::Value)'
../../lib/libjsonrpccpp-common.dll.a(d001120.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x2fe0): multiple definition of `Json::Value::operator!=(Json::Value const&) const'
../../lib/libjsonrpccpp-common.dll.a(d001418.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x3210): multiple definition of `Json::Value::size() const'
../../lib/libjsonrpccpp-common.dll.a(d001374.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x3900): multiple definition of `Json::Value::operator[](std::string const&) const'
../../lib/libjsonrpccpp-common.dll.a(d001413.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x3b70): multiple definition of `Json::Value::isMember(char const*) const'
../../lib/libjsonrpccpp-common.dll.a(d001400.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x3ba0): multiple definition of `Json::Value::isMember(std::string const&) const'
../../lib/libjsonrpccpp-common.dll.a(d001402.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x3bd0): multiple definition of `Json::Value::isNull() const'
../../lib/libjsonrpccpp-common.dll.a(d001383.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x3bf0): multiple definition of `Json::Value::isInt() const'
../../lib/libjsonrpccpp-common.dll.a(d001379.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x40b0): multiple definition of `Json::Value::isString() const'
../../lib/libjsonrpccpp-common.dll.a(d001404.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x40c0): multiple definition of `Json::Value::isArray() const'
../../lib/libjsonrpccpp-common.dll.a(d001388.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x40d0): multiple definition of `Json::Value::isObject() const'
../../lib/libjsonrpccpp-common.dll.a(d001403.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x4700): multiple definition of `Json::Value::toStyledString() const'
../../lib/libjsonrpccpp-common.dll.a(d001366.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x6320): multiple definition of `Json::Value::asInt() const'
../../lib/libjsonrpccpp-common.dll.a(d001376.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x6ef0): multiple definition of `Json::Value::asString() const'
../../lib/libjsonrpccpp-common.dll.a(d001397.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x7780): multiple definition of `Json::Value::~Value()'
../../lib/libjsonrpccpp-common.dll.a(d001118.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x7ad0): multiple definition of `Json::Value::Value(Json::Value const&)'
../../lib/libjsonrpccpp-common.dll.a(d001098.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0x9ed0): multiple definition of `Json::Value::operator[](unsigned int)'
../../lib/libjsonrpccpp-common.dll.a(d001125.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0xa950): multiple definition of `Json::Value::append(Json::Value const&)'
../../lib/libjsonrpccpp-common.dll.a(d001074.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0xb970): multiple definition of `Json::Value::operator[](char const*)'
../../lib/libjsonrpccpp-common.dll.a(d001121.o):(.text+0x0): first defined here
/usr/local/lib/libjsoncpp.a(json_value.cpp.o):json_value.cpp:(.text+0xb9a0): multiple definition of `Json::Value::operator[](std::string const&)'
../../lib/libjsonrpccpp-common.dll.a(d001123.o):(.text+0x0): first defined here
collect2: error: ld returned 1 exit status
src/jsonrpccpp/CMakeFiles/jsonrpcclient.dir/build.make:191: recipe for target 'bin/cygjsonrpccpp-client-0.dll' failed
make[2]: *** [bin/cygjsonrpccpp-client-0.dll] Error 1
CMakeFiles/Makefile2:144: recipe for target 'src/jsonrpccpp/CMakeFiles/jsonrpcclient.dir/all' failed
make[1]: *** [src/jsonrpccpp/CMakeFiles/jsonrpcclient.dir/all] Error 2
Makefile:146: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by cinemast** [comment]
Well, windows is not the primary platform of this project, it is more a side effect.

**Comment by TrevLite** [comment]
I tried that but my program crashes. I think it wants to use an incompatible version of the standard library dll. the crash is that it tries to deference the port number.

**Comment by TrevLite** [comment]
does jsonrpccpp work with the latest version of microhttpd and jsoncpp?

**Comment by TrevLite** [comment]
Ok, I tried to use the pre-build binaries.
The stub generator could not find libstdc++-6.dll but i just copied it into bin.
I tried to compile the example server given in the readme, but i got an error message.
I compiled jsoncpp myself so it might be an ABI incompatability.
Here is my error message:

$ g++ main.cpp -I"/cygdrive/c/Program Files (x86)/libjson-rpc-cpp 0.5.0/include" -L"/cygdrive/c/Program Files (x86)/libjson-rpc-cpp 0.5.0/lib" -lcurl -lmicrohttpd -ljsoncpp -llibjsonrpccpp-common -llibjsonrpccpp-server
/tmp/ccba1mBK.o:main.cpp:(.text$_ZN18AbstractStubServer9sayHelloIERKN4Json5ValueERS1_[__ZN18AbstractStubServer9sayHelloIERKN4Json5ValueERS1_]+0x71): undefined reference to `Json::Value::operator=(Json::Value const&)'
collect2: error: ld returned 1 exit status

**Comment by TrevLite** [comment]
I run it with the debug libstdc++-6.dll and got a more detailed error

(gdb) bt
#0  0x4ec1dddc in std::string::_M_data() const ()
   from /home/Trev_lite/build/test/bin/libstdc++-6.dll
#1  0x4ec1dd0f in std::string::_M_rep() const ()
   from /home/Trev_lite/build/test/bin/libstdc++-6.dll
#2  0x4ec4488b in std::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::string const&) ()
   from /home/Trev_lite/build/test/bin/libstdc++-6.dll
#3  0x68b448b3 in jsonrpc::HttpServer::HttpServer(int, std::string const&, std::string const&, int) ()
   from /home/Trev_lite/build/test/bin/libjsonrpccpp-server.dll
#4  0x004012c8 in main () at main.cpp:32
(gdb) info registers
eax            0x20bf   8383
ecx            0x20bf   8383
edx            0x20bf   8383
ebx            0x80018000       -2147385344
esp            0x28caf8 0x28caf8
ebp            0x28cafc 0x28cafc
esi            0x8003a3b8       -2147245128
edi            0x8003a3b8       -2147245128
eip            0x4ec1dddc       0x4ec1dddc <std::string::_M_data() const+12>
eflags         0x10202  [ IF RF ]
cs             0x23     35
ss             0x2b     43
ds             0x2b     43
es             0x2b     43
fs             0x53     83
gs             0x2b     43

**Comment by cinemast** [comment]
It seems you don't have MinGW in your PATH variable. You should not need to copy libstd*.dll to the bin directory. 

Which Version of MinGW are you using?
Why did you compile libjsoncpp yourself? The latest version of libjsoncpp is completly different so please use the precompiled version. 

Greetings

**Comment by TrevLite** [comment]
I am using Cygwin
I thought that it was a replacement for MinGW
Do I need to install MinGW?

**Comment by cinemast** [comment]
Try it with MinGW. Sorry I am really very fuzzy on that windows stuff.

**Comment by cinemast** [comment]
Any success yet?

**Comment by TrevLite** [comment]
no success
i think i am now having conflicts between mingw and cygwin
i moved to a virtual linux machine but am now having .so instead of .dll problems but it is an improvement so far

**Comment by cinemast** [comment]
Which distro in which version? The major linux distributions really should not make any problems. About the conflicts, why not uninstall cygwin?

**Comment by TrevLite** [comment]
i got it to work on the linux virtual machine. just had to change ld.so.conf
on my windows machines i use cygwin too much to unistall it. MinGW is in my path and is before cygwin (since when i run cygwin it adds itself to the path this causes no problems).

edit: after playing with the path order and compiling with mingw-gcc i got it to only get errors relating to microhttpd and curl with no errors about jsoncpp or jsonrpccpp
atleast i know what to do about this
once i get it to compile i will report back what errors running the program makes if any

**Comment by TrevLite** [comment]
OK got it to work

file this under a request for Cygwin compatibility
I recommend that it says that it does not work with Cygwin and requires MinGW
If it said that i would have saved me a lot of time

**Comment by cinemast** [comment]
I do not intent to personally support any more of windows than there already is. If someone contributes that is fine.

About mentioning mingw: README.md says that is only tested under Windows7 with MinGW32. 

```
Tested under MacOS X (10.9), GNU/Linux (Debian 8 64-bit), Windows 7 (MinGW32) and Raspbian Wheezy (armhf).
```

**Comment by JasonCoombs** [comment]
I just encountered the same problem and solved it definitively with a single change to a CMakeLists.txt file. The problem detailed above is a result of CMake switching to "Library dependencies are transitive by default" which apparently was introduced by policy CMP0022

https://cmake.org/cmake/help/v3.0/command/target_link_libraries.html

We need to insert LINK_PRIVATE into the target_link_libraries( ) line for the jsonrpccommon target, so that the libjsoncpp.a library is linked only with jsonrpccommon without being a "transitive dependency" that is going to be linked into each of the jsonrpc targets (client, server, stub). I'll submit a pull request later after I have finished the build all the way to successful completion with MSYS64/MinGW64.
